src/serac/physics/tests/serac_solid_functional.cpp

```diff
@@ -278,8 +277,99 @@ void functional_solid_test_boundary(double expected_disp_norm, TestType test_mod
   EXPECT_NEAR(expected_disp_norm, norm(solid_solver.displacement()), 1.0e-6);
 }
 
+template <int p, int dim>
+void functional_parameterized_solid_test(double expected_disp_norm)
+{
+  MPI_Barrier(MPI_COMM_WORLD);
+
+  int serial_refinement   = 0;
+  int parallel_refinement = 0;
+
+  // Create DataStore
+  axom::sidre::DataStore datastore;
+  serac::StateManager::initialize(datastore, "solid_functional_parameterized_solve");
+
+  static_assert(dim == 2 || dim == 3, "Dimension must be 2 or 3 for solid functional parameterized test");
+
+  // Construct the appropriate dimension mesh and give it to the data store
+  std::string filename =
+      (dim == 2) ? SERAC_REPO_DIR "/data/meshes/beam-quad.mesh" : SERAC_REPO_DIR "/data/meshes/beam-hex.mesh";
+
+  auto mesh = mesh::refineAndDistribute(buildMeshFromFile(filename), serial_refinement, parallel_refinement);
+  serac::StateManager::setMesh(std::move(mesh));
+
+  // Define a boundary attribute set
+  std::set<int> ess_bdr = {1};
+
+  // define the solver configurations
+  const IterativeSolverOptions default_linear_options = {.rel_tol     = 1.0e-6,
+                                                         .abs_tol     = 1.0e-10,
+                                                         .print_level = 0,
+                                                         .max_iter    = 500,
+                                                         .lin_solver  = LinearSolver::GMRES,
+                                                         .prec        = HypreBoomerAMGPrec{}};
+
+  const NonlinearSolverOptions default_nonlinear_options = {
+      .rel_tol = 1.0e-4, .abs_tol = 1.0e-8, .max_iter = 10, .print_level = 1};
+
+  const typename solid_util::SolverOptions default_static = {default_linear_options, default_nonlinear_options};
+
+  // Construct and initialized the user-defined conductivity to be used as a differentiable parameter in
+  // the thermal conduction physics module.
+  FiniteElementState user_defined_shear_modulus(
+      StateManager::newState(FiniteElementState::Options{.order = 1, .name = "parameterized_shear"}));
+
+  user_defined_shear_modulus = 1.0;
+
+  FiniteElementState user_defined_bulk_modulus(
+      StateManager::newState(FiniteElementState::Options{.order = 1, .name = "parameterized_bulk"}));
+
+  user_defined_bulk_modulus = 1.0;
+
+  // Construct a functional-based thermal conduction solver
```

Comment:
structural/nonlinear mechanics solver?

Reply:
Arg! My copy/paste leavings! I thought I got all of them but I'll do another pass.